src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/BlockingCollection.cs

```diff
@@ -734,7 +714,7 @@ private bool TryTakeWithNoTimeValidation([MaybeNullWhen(false)] out T item, int
                             _freeNodes.Release();
                         }
                     }
-                    else if (removeFaulted)
```

Comment:
New condition should be `removeFaulted || !removeSucceeded && !removeFaulted` => `removeFaulted || !removeSucceeded`, but as we already in else block, `removeSucceeded` is definitely `false` and whole condition folds in single `true`